### PR TITLE
Geolocate 

### PR DESCRIPTION
**Pull request by ohnorobo**
trying to fix up tests for #115

### CODE REVIEW

**Comment by ohnorobo** [comment]
new errors:

PhantomJS 1.9.7 (Linux) ui/map loading data config sets up the map object FAILED
    TypeError: 'undefined' is not a function (evaluating 't.addLayer(this)')
        at /home/travis/build/codeforboston/finda/lib/leaflet/leaflet.js:7
        at /home/travis/build/codeforboston/finda/lib/leaflet/L.Control.Locate.js:67
        at /home/travis/build/codeforboston/finda/lib/leaflet/leaflet.js:9
        at /home/travis/build/codeforboston/finda/src/ui/map.js:44
        at /home/travis/build/codeforboston/finda/lib/es5-shim.min.js:1
        at /home/travis/build/codeforboston/finda/lib/jquery-1.10.2.js:5095
        at /home/travis/build/codeforboston/finda/lib/jquery-1.10.2.js:4766
        at /home/travis/build/codeforboston/finda/lib/jquery-1.10.2.js:5007
        at /home/travis/build/codeforboston/finda/lib/jquery-1.10.2.js:5691
        at /home/travis/build/codeforboston/finda/lib/jquery-1.10.2.js:657
        at /home/travis/build/codeforboston/finda/lib/jquery-1.10.2.js:266
        at /home/travis/build/codeforboston/finda/lib/jquery-1.10.2.js:5692
        at /home/travis/build/codeforboston/finda/lib/flight.min.js:2
        at /home/travis/build/codeforboston/finda/test/spec/ui/map_spec.js:22

so somehow the Location control is failing to extend from leaflet, but only in the tests.

The failing test is
'config sets up the map object' in test/spec/ui/map_spec.js

File: src/script.js [hunk 1]
```diff
@@ -22,6 +23,9 @@ require.config({
       'bootstrap': {
         deps: ['jquery'],
         exports: '$'
+      },
+      'L.Control.Locate': {
```
Comment (paulswartz) [inline src/script.js:27]: Sorry for the long delay; you need to add this section to the test/runner.js file as well.

**Comment by ohnorobo** [comment]
:(

**Comment by paulswartz** [comment]
Ah, figured it out:

``` diff
diff --git a/test/spec/ui/map_spec.js b/test/spec/ui/map_spec.js
index c842cff..99f99a2 100644
--- a/test/spec/ui/map_spec.js
+++ b/test/spec/ui/map_spec.js
@@ -16,8 +16,11 @@ define(
       it('config sets up the map object', function() {
         this.component.map = jasmine.createSpyObj('Map',
                                                   ['setView',
-                                                   'setMaxBounds',
-                                                   'locate']);
+                                                   'setMaxBounds']);
+        spyOn(L.control, 'locate').andReturn(
+          jasmine.createSpyObj('Locate',
+                               ['addTo']));
+
         this.component.map.options = {};
         this.component.trigger('config', mock.config);
         expect(this.component.map.options.maxZoom, mock.config.map.maxZoom);
@@ -25,10 +28,9 @@ define(
           mock.config.map.center, mock.config.map.zoom);
         expect(this.component.map.setMaxBounds).toHaveBeenCalledWith(
           mock.config.map.maxBounds);
-        expect(this.component.map.locate).toHaveBeenCalledWith({
-          setView: true,
-          maxZoom: mock.config.map.zoom
-        });
+        expect(L.control.locate).toHaveBeenCalledWith();
+        expect(L.control.locate().addTo).toHaveBeenCalledWith(
+          this.component.map);
       });

       it('data sets up the features', function() {
```

The `map` in the test is a mocked object, but the Locate control was trying to interact with it like a real map.

**Comment by ohnorobo** [comment]
And there was great rejoicing. Thanks @paulswartz.

**Pull request merged**